docs/environment-and-software/user-environment/containers/examples.md

```diff
@@ -132,6 +132,646 @@ Optimized Tensorflow and PyTorch models are available directly from the NGC.
+        WRF-nvhpc-dev>
+        ```
+        produces an interactive shell environment with access to the GLADE file systems and the containerized build environment.
+
+### Running the container on Derecho
+ The PBS job script listed below shows the steps required to "bind" the host MPI into the container and launch an executable in a batch environment.
+
+???+ example "Containerized WRF PBS Script"
+    === "Intel"
+        ```bash title="run_wrf_intel_container.pbs" linenums="1"
+        ---8<--- "https://raw.githubusercontent.com/NCAR/hpc-demos/main/containers/tutorial/apptainer/WRF/run_wrf_intel_container.pbs"
+        ```
+
+    === "GCC"
+        ```bash title="run_wrf_gcc_container.pbs" linenums="1"
+        ---8<--- "https://raw.githubusercontent.com/NCAR/hpc-demos/main/containers/tutorial/apptainer/WRF/run_wrf_gcc_container.pbs"
+        ```
+
+    === "NVHPC"
+        ```bash title="run_wrf_nvhpc_container.pbs" linenums="1"
+        ---8<--- "https://raw.githubusercontent.com/NCAR/hpc-demos/main/containers/tutorial/apptainer/WRF/run_wrf_nvhpc_container.pbs"
+        ```
+
+    **Discussion**
+
+    The PBS script examines the shared library dependencies of the executable using `ldd`, first within the container and then with the host MPI "injected" (as described [here](./working_with_containers.md#running-containerized-mpi-applications)).  This process is often tedious, iterative, and error prone.  As constructed the PBS script can be executed directly (*without* `qsub`) to inspect these results before waiting for batch resources.
+
+    The `mpiexec` command is fairly standard.  Note that we are using it to launch `singularity`, which in turn will start up the `WRF` executable specified on line 20. note in this case the executable is build into the container, however it could also be resident on GLADE, provided it was complied from this same container development environment.
+
+    The `singularity exec` command lines are complex, so let's deconstruct them here:
+
+    - We make use of the `--bind` argument first to mount familiar GLADE file systems within the container,
+    - and again to "inject" the host MPI into the container. The `/run` directory necessity is not immediately obvious but is used by Cray-MPICH as part of the launching process.
+
+    - We also need to use the `--env` to set the `LD_LIBRARY_PATH` inside the image so that the application can find the proper host libraries.  Recall when we built the FastEddy executable in the containerized environment it had no knowledge of these host-specific paths.  Similarly, we use `--env` to set the `LD_PRELOAD` environment variable inside the container.  This will cause a particular Cray-MPICH library to be loaded prior to application initialization.  This step is not required for "bare metal" execution.
+
+    - We set the `MPICH_SMP_SINGLE_COPY_MODE` environment variable to work around an MPI run-time error hat would otherwise appear.
```

Comment:
small typo: hat should be updated to that! This should be updated to :

"    - We set the `MPICH_SMP_SINGLE_COPY_MODE` environment variable to work around an MPI run-time error that would otherwise appear."